x/aswap/model.go

```diff
@@ -72,24 +72,25 @@ func AsSwap(obj orm.Object) *Swap {
 	return obj.Value().(*Swap)
 }
 
-// Bucket is a type-safe wrapper around orm.Bucket
+// Bucket is a type-safe wrapper around orm.BaseBucket
 type Bucket struct {
-	orm.IDGenBucket
+	orm.XIDGenBucket
 }
 
 // NewBucket initializes a Bucket with default name
 //
-// inherit Get and Save from orm.Bucket
+// inherit Get and Save from orm.BaseBucket
 // add Create
 func NewBucket() Bucket {
-	bucket := migration.NewBucket("aswap", BucketName,
-		orm.NewSimpleObj(nil, &Swap{})).
+	bucket := orm.NewBucketBuilder(BucketName, orm.NewSimpleObj(nil, &Swap{})).
```

Comment:
Yeah, builder seems like a nice way to do this. Reminds me a lot of how you assemble a complex object in Java, but that's what we have - a pretty complex struct with a bunch of options.

I can't help but wonder if we can include `WithMigration` in the builder as well. 

One other thing we could do is include all possible bucket modifications into the builder and then try to convert resulting `built` bucket with a helper, there we can check if all the conditions are satisfied. But maybe this just complicates things even more.

Reply:
`WithMigration` causes cyclomatic dependencies unfortunately.